src/diffusers/loaders.py

```diff
@@ -475,22 +407,47 @@ def load_attn_procs(self, pretrained_model_name_or_path_or_dict: Union[str, Dict
                         cross_attention_dim=cross_attention_dim,
                     )
                     attn_processors[key].load_state_dict(value_dict)
+
+            self.set_attn_processor(attn_processors)
         else:
             raise ValueError(
                 f"{model_file} does not seem to be in the correct format expected by LoRA or Custom Diffusion training."
             )
 
         # set correct dtype & device
-        attn_processors = {k: v.to(device=self.device, dtype=self.dtype) for k, v in attn_processors.items()}
-        non_attn_lora_layers = [(t, l.to(device=self.device, dtype=self.dtype)) for t, l in non_attn_lora_layers]
-
-        # set layers
-        self.set_attn_processor(attn_processors)
+        lora_layers_dict = [(t, l.to(device=self.device, dtype=self.dtype)) for t, l in lora_layers_dict]
 
-        # set ff layers
-        for target_module, lora_layer in non_attn_lora_layers:
+        # set lora layers
+        for target_module, lora_layer in lora_layers_dict:
             target_module.set_lora_layer(lora_layer)
 
+    def convert_state_dict_from_old_format(self, state_dict, network_alphas):
```

Comment:
This function converts the "lora attention processor" key format to the "`LoRACompatibleLinear`" format. After that we can always use the `LoRACompatibleLinear` loading method which allows us to remove a significant amount of code above.

Reply:
We need to account for `LoRACompatibleConv` too.

Reply:
I find `old_format` to be slightly misleading. It could mean either the LoRAs when we didn't have text encoder support or the LoRAs that were monkey-patched. Both of them are referred to as the old formats in the codebase. So, maybe "legacy" is better.